specification.md

```diff
@@ -0,0 +1,92 @@
+- A report of our findings and analysis of the dispute game located in the `security_report` directory.
+
+- A simulation of the dispute game located in the `src` directory. We want to run 100M Sims on a (single or many but takes about 45min to compute) connon trace (on average 1.5b instructions) with at least 5 agent behaviors. The [Simulation Specification](#simulation-specification) section defines the simulation details below.
+
+## The Dispute Game
+The fault-proof dispute game is an interactive proof system used to prove the validity of the layer two states on layer one. Any number of players can play the game. The game has a 7-day chess clock and terminates when the clock runs out. Each player moves by making a `Claim` (see dispute game interface) about an index in a cannon execution trace. If no opposing move is made, the claim implicitly resolves to true. Each player must post a bond to make a claim. If a player makes a false claim and the game resolves, their bond is forfeited and paid to the player who made the opposing correct claim.
+
+- [Dispute Game Interface](https://github.com/ethereum-optimism/specs/blob/main/specs/dispute-game-interface.md)
+- [Fault Dispute Game](https://github.com/ethereum-optimism/specs/blob/main/specs/fault-dispute-game.md)
+
+## Arbiter Simulation
+
+Arbiter is an interface over an EVM sandbox built around the rust evm. Arbiter enables agent-based simulations to model the behavior of different externally owned accounts that would interact with a smart contract or system of smart contracts. This allows us to perturb other variables in the system and observe the effects on the system to discover potential attack vectors and anomalous behavior. 
+
+### Simulation Specification
+These are the details of the simulation design.
+
+#### Initial State
+We configure a simulation by defining an initial state of the system we want before iterating agent behavior over the model. Traditionally, we have used deployer-style agents to deploy the smart contracts we want to simulate with the correct initial state. This has historically involved generating rust contract bindings and interfacing with contracts through these bindings.
+
+Many solidity developers deploy their infrastructure with forge scripts. We have been able to grab the state that is initialized by a forge script on an anvil instance and dump it into a JSON file. This JSON file can then be loaded into an Arbiter instance to initialize the state of the system. The benefit of this is that we can initialize the system's state the same way it would be initialized in production. This allows us to uncover any particular bugs that may be present in the forge script. While this is nice with the optimism mono repo (like a million contracts), it does not mean we will use the sol macro for communicating with the contracts (examples in the `src` directory. 
```

Comment:
Interested with this flow as a whole. Using forge scripts in arbiter in general could be really neat.

Still want to use ethers-rs bindings since our middleware is set up to use those. I'd argue we *need* to use bindings otherwise we are really shooting ourselves in the foot.

Reply:
So take a look at how things are set up with the sendtx macro that ben made here. We are set up to not need to use them at all.

Reply:
This is worth more discussion for sure.